src/hazardous/mac/poly1305.rs

```diff
@@ -399,7 +398,7 @@ impl Poly1305 {
 	}
 }
 
-#[must_use]
+#[must_use = "SECURITY WARNING: Ignoring a Result can have real security implications."]
 /// Initialize a `Poly1305` struct with a given one-time key.
 pub fn init(one_time_key: &OneTimeKey) -> Poly1305 {
```

Comment:
This function does not return a `Result`, so the use of `#[must_use]` should be removed.